api/v1alpha1/api.go

```diff
@@ -144,3 +150,120 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// BackendSecurityPolicyType specifies the type of auth mechanism used to access a backend.
+type BackendSecurityPolicyType string
+
+const (
+	BackendSecurityPolicyTypeAPIKey BackendSecurityPolicyType = "APIKey"
+	BackendSecurityPolicyTypeAWSIAM BackendSecurityPolicyType = "AWS_IAM"
+)
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicy specifies configuration for authentication and authorization rules on the traffic
+// exiting the gateway to the backend.
+type BackendSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              BackendSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// BackendSecurityPolicySpec specifies authentication rules on access the provider from the Gateway.
+type BackendSecurityPolicySpec struct {
+	// Type specifies the auth mechanism used to access the provider. Currently, only "APIKey", AND "AWS_IAM" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;AWS_IAM
+	Type BackendSecurityPolicyType `json:"type"`
+
+	// APIKey is a mechanism to access a backend(s). The API key will be injected into the Authorization header.
+	//
+	// +optional
+	APIKey *AuthenticationAPIKey `json:"apiKey,omitempty"`
+
+	// CloudProviderCredentials is a mechanism to access a backend(s). Cloud provider specific logic will be applied.
+	//
+	// +optional
+	CloudProviderCredentials *AuthenticationCloudProviderCredentials `json:"cloudProviderCredentials,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// BackendSecurityPolicyList contains a list of BackendSecurityPolicy
+type BackendSecurityPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []BackendSecurityPolicy `json:"items"`
+}
+
+// AuthenticationAPIKey specifies the API key.
+type AuthenticationAPIKey struct {
+	// SecretRef is the reference to the secret containing the API key.
+	// ai-gateway must be given the permission to read this secret.
+	// The key of the secret should be "apiKey".
+	//
+	// +optional
+	SecretRef *gwapiv1.SecretObjectReference `json:"secretRef"`
+}
+
+// AuthenticationCloudProviderCredentials specifies supported cloud provider authentication methods
+type AuthenticationCloudProviderCredentials struct {
+	AWSCredentials AWSCredentials `json:"awsCredentials"`
+}
+
+// AWSCredentials contains the supported authentication mechanisms to access aws
+type AWSCredentials struct {
+	// Region specifies the AWS region associated with the policy.
+	//
+	// +kubebuilder:validation:MinLength=1
+	Region string `json:"region"`
+
+	// CredentialsFile specifies the credentials file to use for the AWS provider.
+	//
+	// +optional
+	CredentialsFile *AWSCredentialsFile `json:"credentialsFile,omitempty"`
```

Comment:
Should a SecretRef be used instead of a loacal file here?

The SecretRef appraoch is easier to deploy and more secure compared with a file.